### PR TITLE
More blame types

### PR DESCRIPTION
**Pull request by gl-rockx**
* add implementation for blame type FailedEcies

* add spec test for blame type FailedEcies

* add implementation for blame type InvalidScaler

* add spec test for blame type invalid scaler

* add implementation for invalid commitment blame

* add spec test for blame type InvalidCommitment

* fix lint issues

* remove redundant code from blame spec test

* fix generate json for frost spec tests

* Fix typo

* Fix blame

* Refactor blame

* BlameMessage has to contain SignedMessage

* Refactor blame

* Linting

* Improve readability

* Fix unhandled error

Co-authored-by: gldeng <realgldeng@gmail.com>

### CODE REVIEW

File: qbft/spectest/generate/main.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func writeJson(data []byte) {
 	fullPath := basedir + "/" + fileName
 
 	fmt.Printf("writing spec tests json to: %s\n", fullPath)
-	if err := ioutil.WriteFile(fullPath, data, 0644); err != nil {
+	if err := os.WriteFile(fullPath, data, 0644); err != nil {
```
Comment (alonmuroch) [inline qbft/spectest/generate/main.go:44]: why do we need to change this?

File: ssv/spectest/generate/main.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ func writeJson(data []byte) {
 	fullPath := basedir + "/" + fileName
 
 	fmt.Printf("writing spec tests json to: %s\n", fullPath)
-	if err := ioutil.WriteFile(fullPath, data, 0644); err != nil {
+	if err := os.WriteFile(fullPath, data, 0644); err != nil {
```
Comment (alonmuroch) [inline ssv/spectest/generate/main.go:44]: why change this?

File: ssv/spectest/run_test.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ func TestJson(t *testing.T) {
 	path := filepath.Join(basedir, "generate")
 	fileName := "tests.json"
 	untypedTests := map[string]interface{}{}
-	byteValue, err := ioutil.ReadFile(path + "/" + fileName)
+	byteValue, err := os.ReadFile(path + "/" + fileName)
```
Comment (alonmuroch) [inline ssv/spectest/run_test.go:37]: why change this?

**Review by alonmuroch — CHANGES_REQUESTED** [review]

File: qbft/spectest/generate/main.go (hunk 1 above)
Comment (aayush-rockx) [reply qbft/spectest/generate/main.go:44]: ```Deprecated: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details.```
https://pkg.go.dev/io/ioutil

**Comment by alonmuroch** [comment]
@gl-rockx could you rebase so we can merge?

**Pull request merged**